### PR TITLE
Added OpenShift deployment configuration

### PR DESCRIPTION
**Pull request by skabashnyuk**
### What does this PR do?
Added OpenShift deployment configuration

### Reference Issue
https://github.com/eclipse/che/issues/10652

### CODE REVIEW

File: openshift/os-che-plugin-registry.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+---
+apiVersion: v1
+kind: Template
+metadata:
+  name: os-che-plugin-registry-service
```
Comment (riuvshin) [inline openshift/os-che-plugin-registry.yml]: I would avoid of `os-` prefix here

File: openshift/os-che-plugin-registry.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  spec:
+    host: ${APPLICATION_DOMAIN}
+    to:
+      kind: Service
+      name: ${NAME}
+- apiVersion: v1
+  kind: DeploymentConfig
+  metadata:
+    labels:
+      app: os-che-plugin-registry
```
Comment (riuvshin) [inline openshift/os-che-plugin-registry.yml]: and here

File: openshift/os-che-plugin-registry.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    ports:
+      - protocol: TCP
+        port: 80
+        targetPort: 8080
+    selector:
+      deploymentconfig: ${NAME}
+parameters:
+- name: NAME
+  required: true
+  value: che-plugin-registry
```
Comment (riuvshin) [inline openshift/os-che-plugin-registry.yml]: name is not something we will change :) why it is configurable?

File: openshift/os-che-plugin-registry.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  value: latest
+- description: Maximum amount of memory the container can use.
+  displayName: Memory Limit
+  name: MEMORY_LIMIT
+  required: true
+  value: 256Mi
+- description: The exposed hostname that will route to the httpd service, if left
+    blank a value will be defaulted.
+  displayName: Application Hostname
+  name: APPLICATION_DOMAIN
```
Comment (riuvshin) [inline openshift/os-che-plugin-registry.yml]: new line plese

**Comment by riuvshin** [comment]
like approve very good spasiba klass!

File: openshift/che-plugin-registry.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      rollingParams:
+        intervalSeconds: 1
+        maxSurge: 25%
+        maxUnavailable: 25%
+        timeoutSeconds: 600
+        updatePeriodSeconds: 1
+      type: Rolling
+    template:
+      metadata:
+        creationTimestamp: null
```
Comment (sleshchenko) [inline openshift/che-plugin-registry.yml]: Could you explain why do we need this?

Comment (skabashnyuk) [reply openshift/che-plugin-registry.yml]: I'll remove it

File: openshift/che-plugin-registry.yml [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+kind: Template
+metadata:
+  name: che-plugin-registry-service
+objects:
+- apiVersion: v1
+  kind: Route
+  metadata:
+    name: che-plugin-registry
+  spec:
+    host: ${APPLICATION_DOMAIN}
```
[the remark this answers is not in the document]
Comment (riuvshin) [reply openshift/che-plugin-registry.yml]: I had same questiong

**Pull request merged**